taichi/gui/gui.h

```diff
@@ -503,10 +503,19 @@ class GUI : public GUIBase {
   };
 
   struct KeyEvent {
-    enum class Type { move, press, release };
+    enum class Type { scroll = 1, move, press, release };
     Type type;
     std::string key;
-    Vector2i pos;
+    Vector2i data;
```

Comment:
We may use two vector to store pos and delta? So that user can get exactly the position of cursor when wheel happens, this could be useful when implenenting scaling centered at cursor.
The delta is simply wasted when it's not a scroll event.

Reply:
Agree . 
Reserving bytes for some additional info is a usual way to implement . Just add it if we need this position info indeed while handling wheel event . 8 bytes waste won't cause performance problem at present.